opencensus/trace/propagation/b3_format.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2018, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from opencensus.trace.span_context import SpanContext, INVALID_SPAN_ID
+from opencensus.trace.trace_options import TraceOptions
+
+_TRACE_ID_KEY = 'x-b3-traceid'
+_SPAN_ID_KEY = 'x-b3-spanid'
+_SAMPLED_KEY = 'x-b3-sampled'
+
+
+class B3FormatPropagator(object):
+    """Propagator for the B3 HTTP header format.
+
+    See: https://github.com/openzipkin/b3-propagation
+    """
+
+    def from_headers(self, headers):
+        """Generate a SpanContext object from B3 propagation headers.
+
+        :type headers: dict
+        :param headers: HTTP request headers.
+
+        :rtype: :class:`~opencensus.trace.span_context.SpanContext`
+        :returns: SpanContext generated from B3 propagation headers.
+        """
+        if headers is None:
+            return SpanContext(from_header=False)
+
+        trace_id = headers.get(_TRACE_ID_KEY)
+        span_id = headers.get(_SPAN_ID_KEY)
+        sampled = headers.get(_SAMPLED_KEY) == '1'
```

Comment:
the header can be absent. when it is it should be sampled using whatever your heuristic is not set to unsampled. make sense?

Reply:
From what I understand this will work if it's missing, because later we call `.enabled or .should_sample` [here](https://sourcegraph.com/github.com/census-instrumentation/opencensus-python@c84e87f/-/blob/opencensus/trace/tracer.py#L76-77)